### PR TITLE
Remove disable option for metrics server for users

### PR DESCRIPTION
**Pull request by louiseschmidtgen**
This PR enables metrics-server by default and ensures that there is no way to disable it. 
Metrics-server is removed from k8s get/set/enable/disable/status commands. Removed asking to enable it on bootstrap. This PR also updates the `test_metrics_server` integration test.

### CODE REVIEW

**Review by bschimke95 — COMMENTED** [review]
The metrics server should be completely removed from the `UserFacingClusterConfig`. Otherwise, the API would still be able to  enable/disable the feature.

File: src/k8s/cmd/k8s/k8s_bootstrap.go [hunk 1]
```diff
@@ -107,13 +107,13 @@ func newBootstrapCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
...
 						},
-						MetricsServer: apiv1.MetricsServerConfig{
-							Enabled: vals.Pointer(true),
-						},
 					},
 				}
 			}
 
+			// Always enable MetricsServer by default
+			bootstrapConfig.ClusterConfig.MetricsServer.Enabled = vals.Pointer(true)
```
Comment (neoaggelos) [inline src/k8s/cmd/k8s/k8s_bootstrap.go:115]:
```suggestion
if bootstrapConfig.ClusterConfig.MetricsServer.Enabled == nil {
			bootstrapConfig.ClusterConfig.MetricsServer.Enabled = vals.Pointer(true)
}
```

File: tests/integration/tests/test_metrics_server.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ def test_metrics_server(instances: List[harness.Instance]):
     instance = instances[0]
     util.wait_for_dns(instance)
     util.wait_for_network(instance)
-    instance.exec(["k8s", "enable", "metrics-server"])
+    # metrics-server is enabled on bootstrap
```
Comment (neoaggelos) [inline tests/integration/tests/test_metrics_server.py:21]: you can just remove the line completely

**Review by eaudetcobello — COMMENTED** [review]
Question - why is this left unchanged? I see that in this PR you're setting it to enabled by default on bootstrap, but why wouldn't this also change? https://github.com/canonical/k8s-snap/blob/6eb114f1dceac20af3863b30dca8de96e210d274/src/k8s/pkg/k8sd/types/cluster_config_defaults.go#L97

**Comment by louiseschmidtgen** [comment]
> Question - why is this left unchanged? I see that in this PR you're setting it to enabled by default on bootstrap, but why wouldn't this also change?
> 
> https://github.com/canonical/k8s-snap/blob/6eb114f1dceac20af3863b30dca8de96e210d274/src/k8s/pkg/k8sd/types/cluster_config_defaults.go#L97

For any "way" (file/cmd/input) that we can bootstrap the cluster we explicitly enable enable metrics-server in `hooks_bootstrap`, but you are right you would expect the function to do that as well. I added it for completeness 👍🏼

File: src/k8s/pkg/k8sd/api/cluster.go [hunk 3]
```diff
@@ -50,5 +50,8 @@ func (e *Endpoints) getClusterStatus(s *state.State, r *http.Request) response.R
 		},
 	}
 
+	// Hide metrics-server in the cluster status response
+	result.ClusterStatus.Config.MetricsServer.Enabled = nil
+
```
Comment (neoaggelos) [inline src/k8s/pkg/k8sd/api/cluster.go:55]: let's revert this line, and instead ignore it on the client side:

In https://github.com/canonical/k8s-snap/blob/ffb494961acf113a2af715a35e634c8b6a522a53/src/k8s/cmd/k8s/k8s_get.go#L29-L36

```go
			// ignore MetricsServer config
			config.MetricsServer = apiv1.MetricsServerConfig{}
```

- In https://github.com/canonical/k8s-snap/blob/ffb494961acf113a2af715a35e634c8b6a522a53/src/k8s/cmd/k8s/k8s_status.go#L31-L37

```go

			// ignore MetricsServer config
			status.Config.MetricsServer = apiv1.MetricsServerConfig{}
```

File: src/k8s/cmd/k8s/k8s_set.go [hunk 4]
```diff
@@ -171,12 +171,6 @@ func updateConfig(config *apiv1.UserFacingClusterConfig, arg string) error {
 			return fmt.Errorf("invalid integer value for load-balancer.bgp-peer-asn: %w", err)
 		}
 		config.LoadBalancer.BGPPeerASN = &v
-	case "metrics-server.enabled":
-		v, err := strconv.ParseBool(value)
-		if err != nil {
-			return fmt.Errorf("invalid boolean value for metrics-server.enabled: %w", err)
-		}
-		config.MetricsServer.Enabled = &v
 	default:
```
Comment (neoaggelos) [inline src/k8s/cmd/k8s/k8s_set.go:174]: we can revert this

File: src/k8s/cmd/k8s/k8s_enable.go [hunk 5]
```diff
@@ -64,10 +64,6 @@ func newEnableCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 					config.LoadBalancer = api.LoadBalancerConfig{
 						Enabled: vals.Pointer(true),
 					}
-				case "metrics-server":
-					config.MetricsServer = api.MetricsServerConfig{
-						Enabled: vals.Pointer(true),
-					}
 				}
```
Comment (neoaggelos) [inline src/k8s/cmd/k8s/k8s_enable.go:67]: we can revert this

File: src/k8s/cmd/k8s/k8s_disable.go [hunk 6]
```diff
@@ -64,10 +64,6 @@ func newDisableCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 					config.LoadBalancer = api.LoadBalancerConfig{
 						Enabled: vals.Pointer(false),
 					}
-				case "metrics-server":
-					config.MetricsServer = api.MetricsServerConfig{
-						Enabled: vals.Pointer(false),
-					}
 				}
```
Comment (neoaggelos) [inline src/k8s/cmd/k8s/k8s_disable.go:67]: we can revert this

File: src/k8s/cmd/k8s/k8s_bootstrap.go [hunk 7]
```diff
@@ -108,13 +108,15 @@ func newBootstrapCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
...
-						},
 					},
 				}
 			}
 
+			// Always enable MetricsServer by default
+			if bootstrapConfig.ClusterConfig.MetricsServer.Enabled == nil {
+				bootstrapConfig.ClusterConfig.MetricsServer.Enabled = vals.Pointer(true)
+			}
+
```
Comment (neoaggelos) [inline src/k8s/cmd/k8s/k8s_bootstrap.go:119]: we can revert this, as it is handled in https://github.com/canonical/k8s-snap/pull/327/files#diff-b5b56d715e342b6e5c59f5158d6cf255bf5f1ff11f6d7ff786f6d98b8842eb1bR98

**Review by neoaggelos — COMMENTED** [review]
Almost there, just a couple minor fixes

**Review by neoaggelos — APPROVED** [review]

File: src/k8s/cmd/k8s/k8s_get.go [hunk 8]
```diff
@@ -27,13 +27,16 @@ func newGetCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
...
 
-			config, err := client.GetClusterConfig(cmd.Context(), api.GetClusterConfigRequest{})
+			config, err := client.GetClusterConfig(cmd.Context(), apiv1.GetClusterConfigRequest{})
 			if err != nil {
 				cmd.PrintErrf("Error: Failed to get the current cluster configuration.\n\nThe error was: %v\n", err)
 				env.Exit(1)
 				return
 			}
 
+			// ignore MetricsServer config
```
Comment (bschimke95) [inline src/k8s/cmd/k8s/k8s_get.go:37]: nit: this commment doesn't really add any context. IMHO, either remove it or add some context on *why* this is ignored.

**Review by bschimke95 — APPROVED** [review]
LGTM, only a minor nit

**Pull request merged**